src/source/PeerConnection/PeerConnection.c

```diff
@@ -184,7 +184,10 @@ STATUS sendPacketToRtpReceiver(PKvsPeerConnection pKvsPeerConnection, PBYTE pBuf
             CHK(NULL != (pPayload = (PBYTE) MEMALLOC(bufferLen)), STATUS_NOT_ENOUGH_MEMORY);
             MEMCPY(pPayload, pBuffer, bufferLen);
 
-            CHK_STATUS(createRtpPacketFromBytes(pPayload, bufferLen, &pRtpPacket));
+            if (STATUS_FAILED(retStatus = createRtpPacketFromBytes(pPayload, bufferLen, &pRtpPacket))) {
+                MEMFREE(pPayload);
+                CHK_LOG_ERR_NV(retStatus);
+            }
             CHK_STATUS(jitterBufferPush(pTransceiver->pJitterBuffer, pRtpPacket));
```

Comment:
You could have a flag controlling the freeing of the pPayload. If after pushing something fails, it won't free